numpy/testing/_private/utils.py

```diff
@@ -788,32 +789,34 @@ def func_assert_same_pos(x, y, func=isnan, hasval='nan'):
         # np.ma.masked, which is falsy).
         if cond != True:
             n_mismatch = reduced.size - reduced.sum(dtype=intp)
-            percent_mismatch = 100 * n_mismatch / ox.size
+            n_elements = flagged.size if flagged.ndim != 0 else reduced.size
+            percent_mismatch = 100 * n_mismatch / n_elements
             remarks = [
                 'Mismatched elements: {} / {} ({:.3g}%)'.format(
-                    n_mismatch, ox.size, percent_mismatch)]
+                    n_mismatch, n_elements, percent_mismatch)]
 
             with errstate(invalid='ignore', divide='ignore'):
                 # ignore errors for non-numeric types
                 with contextlib.suppress(TypeError):
                     error = abs(x - y)
-                    max_abs_error = error.max()
-                    if error.dtype == 'object':
+                    max_abs_error = np.max(error)
+                    if hasattr(error, 'dtype') and error.dtype != 'object':
```

Comment:
Nit: Perhaps `getattr(error, 'dtype', np.object_) == np.object_` which avoids repeated lookups or string coercion - and also minimizes the diff compared to the original